backend/src/main/java/eu/clarin/switchboard/resources/ToolsResource.java

```diff
@@ -42,4 +46,46 @@ public Response getTools(@QueryParam("mediatype") String mediatype,
         tools.sort((t1, t2) -> t1.getName().compareToIgnoreCase(t2.getName()));
         return Response.ok(tools).build();
     }
+
+    @GET
+    @Path("logos/{logoName}")
+    public Response getTools(@PathParam("logoName") String logoName) throws IOException {
+        String imageMimeType = URLConnection.guessContentTypeFromName(logoName);
+        java.nio.file.Path logo = Paths.get(toolConfig.getLogoRegistryPath(), logoName);
+
+        byte[] data;
+        if (logo.toFile().exists()) {
+            data = Files.readAllBytes(logo);
+        } else {
+            LOGGER.debug("logo not found, trying to find it in resources: " + logoName);
+            data = this.readSmallResource(logoName);
+        }
+
+        if (data == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
```

Comment:
I suggest using `Optional` here, using `.ofNullable` for `Files.readAllBytes`, returning a native optional from `readSmallResource` and then map the result to a response (using orElse for the not found case)

Reply:
I don't think that will improve the readability of the code, so I'll keep it as it is. I think I still have to check for the existence of the file, otherwise we get an exception.

Reply:
Ok, it might not depending on who's reading :) Reducing semantic nulls and the need for null-checks is something I generally tend to strive for now Java offers this, but in any case there are no real strong objective arguments either way and of course it's your call :)